tests/x11/ghostscript.pm

```diff
@@ -31,26 +31,24 @@ use testapi;
 use utils;
 
 sub run {
-    select_console "x11";
-    x11_start_program('xterm');
+    my $self = shift;
+    $self->select_serial_terminal;
 
-    # become root, disable packagekit and install all needed packages
-    become_root;
+    # disable packagekit and install all needed packages
     quit_packagekit;
     zypper_call "in ghostscript ghostscript-x11";
 
     # special case for gv which is not installed on all flavors
     my $gv_missing = zypper_call("in gv", exitcode => [0, 104]);
-
-    # exit root shell
-    type_string "exit\n";
-
-    my $gs_script = "ghostscript_ps2pdf.sh";
-    my $gs_log    = "ghostscript.log";
-    my $gs_failed = "/tmp/ghostscript_failed";
-    my $reference = "alphabet.pdf";
+    my $gs_script  = "ghostscript_ps2pdf.sh";
+    my $gs_log     = "ghostscript.log";
+    my $gs_failed  = "/tmp/ghostscript_failed";
+    my $reference  = "alphabet.pdf";
 
     # download ghostscript converter script and test if download succeeded
+    # run script as default user to create files which can be accessed later
+    type_string "su $username\n";
+    type_string "cd --\n";
```

Comment:
Following @mdoucha's last suggestion:

```suggestion
   my $working_directory = script_output("mktemp -d -t GHOSTSCRIPTXXX");
    type_string "$working_directory\n";
```

Reply:
Later before changing to x11, do:

```
script_run "chmod -R $username $working_directory"

Reply:
once xterm is open, `cd $working_directory`

Reply:
That's quite a lot of overkill. Simple `assert_script_run('cd ~' . $testapi::username);` will suffice. The `rm -f` at the end will remove files owned by root just fine. Once the missing virtio terminal gets added, we can simply swap the `cd` for `select_user_serial_terminal()`.